[FEATURE] Bug reporting

juanyepesp opened the pull request:
CLOSING ISSUES
- closes #165 

DESCRIPTION
- This pull request implements a 🆕 view for creating a bug report. Right now, the view is inside the `UserView`, which is temporary. It should go inside settings.
- The information is sent to firebase using a Task and a fire and forget strategy so the UI isn't blocked by this operation. (Just like @mar-cas3 with Review reporting)
- Bug report can not be sent without connection. 

TODO:
- [x] Button is disabled but is not showing gray 
- [x] Lost connection EC message
- [x] Drafting feature (using the same strategy as review drafting)
- [x] adding a toolbar like the one in the create review and review report views
- [x] add the date of the report in the data being sent to firebase
- [x] change from user view to settings view once merged



DEMO:

https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/77981728/aad35e60-8fbb-47af-b5f5-fdb6a8b2ac09


Feature by @juanyepesp & @Laurarestrepo03

mar-cas3 committed:
feat(TextFieldWithLimit.swift): add text field with char limit and count

Signed-off-by: Maria Castro Iregui <m.castroi@uniandes.edu.co>

mar-cas3 committed:
fix(): missing references in project file

Signed-off-by: Maria Castro Iregui <m.castroi@uniandes.edu.co>

juanyepesp committed:
feat: Add UI View

juanyepesp committed:
Merge remote-tracking branch 'origin/feature/text-field-component' into feature/165-bug-reporting-view-with-drafts

juanyepesp committed:
feat: implement bug report view and preliminary functionality

Laurarestrepo03 committed:
feat: bug report draft local db support

Laurarestrepo03 committed:
refactor: comment

Laurarestrepo03 committed:
feat: gray button when no connection + commented stuff

mar-cas3 commented:
@juanyepesp in the data sent to firebase please add the date of the report, or lmk and i can add it

juanyepesp committed:
fix: see description

- adds date to BugReport
- changes bookmarks no bookmarks view message
- adds toolbar
- makes largebutton less annoying when keyboard pops up

juanyepesp committed:
Merge branch 'develop' into feature/165-bug-reporting-view-with-drafts

juanyepesp committed:
fix: adds EC message

- changes text when no bookmarks
- adds save as draft button
- restores UserView from `develop`

juanyepesp committed:
refactor: removed some comments

juanyepesp committed:
fix: EC message centered

Laurarestrepo03 committed:
feat: bug report drafts ui

Laurarestrepo03 commented:
**Commit 6b96c209d99723e97b63e4d928d12126c988e2af:**

- Adds UI for bug report drafts
- Flows:
   - Flow 1: If draft exists when opening view, alert will ask user if they want to load it > user selects yes > fields are changed to draft values > user submits bug report > draft is deleted
   - Flow 2: User opens view > user taps 'Save as draft' > alert asks user if they want to save as draft (this will delete the previous one) > user selectes yes > if previous draft exists it is deleted, new draft is saved in the DB
- Currently, if the user chooses to save the draft, the view isn't dismissed. I could add it, but I think it's okay to leave it like this since they can go back manually

Alerts:

<img alt="load draft alert" height="400" src="https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/69609680/65ce1b07-cc50-4692-990d-9aac5e7124cf">
<img alt="save draft alert" height="400" src="https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/69609680/49824da3-ad0f-45ac-a132-643138a7524a">

Laurarestrepo03 committed:
Merge branch 'develop'

Conflicts:
	FoodBookApp.xcodeproj/project.pbxproj
	FoodBookApp/UI/Views/User/UserView.swift

Laurarestrepo03 committed:
fix: bundle identifier

mar-cas3 reviewed (approved):
great job guys!
<img src="https://github.com/ISIS3510-202410-Team23/SwiftApp/assets/61592394/f7eb01b2-fc66-46a0-9c59-972c7af9b9ca" width="300">
